### PR TITLE
Signpost energy library updates

### PR DESCRIPTION
**Pull request by adkinsjd**
Update the energy monitor library. Not ready to merge (I'll push some more stylistic fixes), but I wanted to get it in and start high level review.

Summary of changes:
 - rename signpost_energy to signpost_energy_monitors (this is to differentiate between upcoming signpost_energy_policy libraries)
   - also remove signpost_energy.h include from apps that don't use it
   - unfortunately it seems that changes didn't get tracked across rename :disappointed: 
 - A near rewrite of the signpost_energy(_monitors) library. 
    - More consistent naming of functions
    - units removed from names of functions
    - all conversions pushed into drivers
    - return types that better match reality
 - Apps are all updated to take advantages of these changes

### CODE REVIEW

File: apps/libsignpost/signpost_energy_monitors.c [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+static uint8_t module_num_to_selector_mask[8] = {0x4, 0x8, 0x10, 0, 0, 0x20, 0x40, 0x80};
+static ltc294x_model_e model;
+static uint16_t prescaler;
+
+
+////////////////////////////////////////////////////////////
+//Initialization functions
+//Make sure to call the right one
+/////////////////////////////////////////////////////////
+void signpost_energy_init_ltc2941 (void) {
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: I think there should still just be `signpost_energy_init` exposed to apps. We have the ID chip on boards for a reason, the library should handle choosing the right energy monitor for you

File: apps/libsignpost/signpost_energy_monitors.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+static uint8_t module_num_to_selector_mask[8] = {0x4, 0x8, 0x10, 0, 0, 0x20, 0x40, 0x80};
+static ltc294x_model_e model;
+static uint16_t prescaler;
+
+
+////////////////////////////////////////////////////////////
+//Initialization functions
+//Make sure to call the right one
+/////////////////////////////////////////////////////////
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: nit: missing spaces between `//Initialization` (and next line)

File: apps/libsignpost/signpost_energy_monitors.c [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        ltc294x_configure_sync(LTC2943, InterruptPinAlertMode, POWER_MODULE_PRESCALER_LTC2943, ADCScan);
+    }
+
+    model = LTC2943;
+    prescaler = POWER_MODULE_PRESCALER_LTC2943;
+
+    // set all channels open for Alert Response
+    i2c_selector_select_channels_sync(0x1FF);
+
+    //max17205_configure_pack_sync();
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: ?

File: apps/libsignpost/signpost_energy_monitors.c [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+////////////////////////////////////////////////
+// Static helper functions
+// ////////////////////////////////////////////
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: nit: one missing `/`

File: apps/libsignpost/signpost_energy_monitors.c [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+static void reset_ltc_energy (int selector_mask) {
+	// Select correct LTC294x
+    i2c_selector_select_channels_sync(selector_mask);
+
+	ltc294x_reset_charge_sync();
+}
+
+///////////////////////////////////////////////////
+// Energy Reset Functions
+// ////////////////////////////////////////////////
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: `/`

File: apps/libsignpost/signpost_energy_monitors.c [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+void signpost_energy_reset_linux_energy (void) {
+    reset_ltc_energy(0x2);
+}
+
+///////////////////////////////////////////////////
+// Raw coulomb counter read functions
+// ///////////////////////////////////////////////
+uint32_t signpost_energy_get_controller_energy (void) {
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: Why does this not have a unit attached to it any more? The interior function has `uah`, but we're hiding that from the caller? I very consciously put units on every function when I wrote this the first time, it's makes it way more usable
(edit: all of these need units)

File: apps/libsignpost/signpost_energy_monitors.c [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+}
+uint32_t signpost_energy_get_module_energy (int module_num) {
+	return get_ltc_energy_uah(module_num_to_selector_mask[module_num],POWER_MODULE_RSENSE)*MODULE_VOLTAGE;
+}
+
+////////////////////////////////////////////////////
+// Raw current read functions
+// /////////////////////////////////////////////////////
+
+uint32_t signpost_energy_get_controller_current (void) {
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: Units here too

File: apps/libsignpost/signpost_energy_monitors.c [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+uint32_t signpost_energy_get_solar_current (void) {
+    return get_ltc_current_ua(0x100,POWER_MODULE_SOLAR_RSENSE);
+}
+
+int32_t signpost_energy_get_battery_current (void) {
+    uint16_t voltage;
+    int16_t current;
+    max17205_read_voltage_current_sync(&voltage,&current);
+    float c = max17205_get_current_uA(current);
+    return (int)c;
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: return cast doesn't match return type

File: apps/libsignpost/signpost_energy_monitors.c [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+
+///////////////////////////////////////////////
+//More information about the battery
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: nit: missing space

File: apps/libsignpost/signpost_energy_monitors.c [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+//////////////////////////////////////////////
+uint32_t signpost_energy_get_battery_capacity (void) {
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: unit
(and below)

File: apps/libsignpost/signpost_energy_monitors.c [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    uint16_t charge;
+    uint16_t full;
+    max17205_read_soc_sync(&percent, &charge, &full);
+    return max17205_get_capacity_uAh(charge)*BATTERY_VOLTAGE_NOM;
+}
+
+
+/////////////////////////////////////////////////////////////////////
+// These functions give you voltages for battery and solar
+// ///////////////////////////////////////////////////////////////////
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: `/`

File: apps/libsignpost/signpost_energy_monitors.c [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+uint16_t signpost_energy_get_battery_voltage (void) {
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.c]: units here too

File: apps/libsignpost/signpost_energy_monitors.h [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// qlsb = 0.0625mAh with 0.017 Ohm sense resistor
+#define POWER_MODULE_PRESCALER_LTC2941 32
+#define POWER_MODULE_PRESCALER_LTC2943 64
+
+//in Mohm
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.h]: style: missing spaces on a bunch of these comments

File: apps/libsignpost/signpost_energy_monitors.h [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+#define POWER_MODULE_RSENSE 17
+#define POWER_MODULE_SOLAR_RSENSE 50
```
Comment (ppannuto) [inline apps/libsignpost/signpost_energy_monitors.h]: Why bury the unit in a comment here? `POWER_MODULE_RSENSE_MOHM`?

**Review by ppannuto — CHANGES_REQUESTED** [review]
Only looked over libsignpost so far, none of the apps

File: apps/libsignpost/signpost_energy_monitors.c (hunk 1 above)
Comment (adkinsjd) [reply apps/libsignpost/signpost_energy_monitors.c]: I totally agree, we just don't have anything working to do that right now

Comment (adkinsjd) [reply apps/libsignpost/signpost_energy_monitors.c]: There needs to be an i2c eeprom driver written - which feels like something that should go in another PR

File: apps/libsignpost/signpost_energy_monitors.c (hunk 3 above)
Comment (adkinsjd) [reply apps/libsignpost/signpost_energy_monitors.c]: What are you questioning? The configure pack? I should remove that - it no longer exists.

File: apps/libsignpost/signpost_energy_monitors.c (hunk 6 above)
Comment (adkinsjd) [reply apps/libsignpost/signpost_energy_monitors.c]: I explicitly removed units, and instead decided to pick units for the whole library (see my summary comments). We can add them back - I don't care that much - but I just thought if they were all the same we didn't need to put the unit name.

There are comments with units in the header.

File: apps/libsignpost/signpost_energy_monitors.h (hunk 14 above)
Comment (adkinsjd) [reply apps/libsignpost/signpost_energy_monitors.h]: Yeah that's a good fix

**Comment by adkinsjd** [comment]
Per my first comment I hadn't done any stylistic passes on this yet so yall don't need to make that effort yet unless you just want to. I appreciate the help, but it feels like a waste of your time until I put at least a little effort into style.

File: apps/libsignpost/signpost_energy_monitors.c (hunk 6 above)
Comment (ppannuto) [reply apps/libsignpost/signpost_energy_monitors.c]: In one case the unit is with the function where it's used every place it's used automatically. In the other case I have to go look up a comment in a header file to see what unit I'm getting back. Having the unit with the code really helps readability

**Comment by ppannuto** [comment]
Missed the parenthetical on style, but I didn't spend too much time thinking about that, just highlighting when I saw them

**Comment by adkinsjd** [comment]
Pat all of your changes have been addressed except automatic configuration (which as discussed should go in another pr after eeprom driver is done)

**Review by ppannuto — APPROVED** [review]
There are several style nits still, but there's a PR working through Tock that will just fix all of those automatically, so not worth fixing here.

I think this is gtg

**Pull request merged**